File: src/database/DBDefaultDeliveryStop.java
package database;

import model.DefaultDeliveryStop;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

/**
 * DBDefaultDeliveryStop
 * Handles all database functionality for default delivery stops.
 *
 * @author Dani Sander
 * @version 1.0
 * @since 20-05-15
 */

public class DBDefaultDeliveryStop {

    private DBConnection dbConnection;
    private static DBDefaultDeliveryStop instance;

    /**
     * Private constructor for singleton.
     *
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    private DBDefaultDeliveryStop() throws ClassNotFoundException, SQLException {
        dbConnection = DBConnection.getInstance();
    }

    /**
     * Singleton method for class.
     *
     * @return instance of class.
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    public static DBDefaultDeliveryStop getInstance() throws ClassNotFoundException, SQLException {
        if (instance == null) {
            instance = new DBDefaultDeliveryStop();
        }

        return instance;
    }


    /**
     * @param defaultRouteID
     * @return list of all DefaultDeliveryStop for the given defaultRouteID
     */
    public ArrayList<DefaultDeliveryStop> getDefaultDeliveryStops(long defaultRouteID) {
        ArrayList<DefaultDeliveryStop> list;
        String sql = String.format("select * from DefaultDeliveryStop where default_route_id = '%s';", defaultRouteID);
        list = (ArrayList<DefaultDeliveryStop>) dbConnection.sendSQL(this, sql, "_formatDefaultDeliveryStop");
        return list;
    }


    /**
     * @param rs takes the ResultSet from database
     * @return list of DefaultDeliveryStop
     */
    public ArrayList<DefaultDeliveryStop> _formatDefaultDeliveryStop(ResultSet rs) {
        ArrayList<DefaultDeliveryStop> tableList = new ArrayList<DefaultDeliveryStop>();
        try {
            while (rs.next()) {
                tableList.add(
                        new DefaultDeliveryStop(
                                rs.getLong("id"),
                                rs.getLong("geo_loc_id")
                        ));
            }
        } catch (SQLException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return tableList;
    }
}

File: Server/Database/DBDefaultDeliveryStop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;
using System.Data;

namespace Server.Database

{
    /// all database functionality for default delivery stops will be handled by this class
    /// 
    /// </summary>
 

    public class DBDefaultDeliveryStop
    {
        public DBConnection DbConnection { get; set; }
        private static DBDefaultDeliveryStop instance; 

        /// <summary >
        /// private constructor for singelton
        /// </summary>
       
        private DBDefaultDeliveryStop()
        {
            DbConnection = DBConnection.Instance;
        }

        /// <sunmmary>
        /// singelton get instance method 
        /// returns the instance from DB 
      
        public static DBDefaultDeliveryStop Instance
        {
            get
            {
                if (instance == null)
                {
                    instance = new DBDefaultDeliveryStop();
                }

                return instance;
            }

        }

        /**
     * @param defaultRouteID
     * @return list of all DefaultDeliveryStop for the given defaultRouteID
     */
        public List<DefaultDeliveryStop> GetDefaultDeliveryStops(long defaultRouteID)
        {
           List<DefaultDeliveryStop> list;
            String sql = String.Format("select * from DefaultDeliveryStop where default_route_id = '{0}';", defaultRouteID);
            list = DbConnection.SendSQL<DefaultDeliveryStop>(sql, ConvertTotDefaultDeliveryStop);
            return list;
        }


        /**
         * @param rs takes the ResultSet from database
         * @return list of DefaultDeliveryStop
         */
        private List<DefaultDeliveryStop> ConvertTotDefaultDeliveryStop(IDataReader dataSet)
        {
           List<DefaultDeliveryStop> tableList = new List<DefaultDeliveryStop>();
                while (dataSet.Read())
                {
                    tableList.Add(new DefaultDeliveryStop(
                                    dataSet.GetInt64(0),
                                    dataSet.GetInt64(1)
                            ));
                }

            return tableList;
        }

    }
}

        


    

